### ISSUE TITLE
pkgrepo.managed key_url fails with HTTP(S) Basic Auth

### ISSUE THREAD

**Issue by SaltwaterC** [saltstack/salt#19728]
Something changed in the way URLs are handled for key_url as in the downloaded key file is the HTML of the HTTP 401 response and apt-key add (obviously) fails. The configured repo is something like:

``` yaml
foo-repo:
  pkgrepo.managed:
    - name: deb https://username:{{ pillar['foo']['repo_http_auth'] }}@repo.hostname foo bar
    - key_url: https://username:{{ pillar['foo']['repo_http_auth'] }}@repo.hostname/public.key
```

The state fails with:

```
Error: key retrieval failed: apt-key add /var/cache/salt/minion/extrn_files/base/username:password@repo.hostname/public.key
```

``` bash
salt --versions-report
           Salt: 2014.7.0
         Python: 2.7.3 (default, Dec 18 2014, 19:10:20)
         Jinja2: 2.6
       M2Crypto: 0.21.1
 msgpack-python: 0.1.10
   msgpack-pure: Not Installed
       pycrypto: 2.4.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 13.0.0
           RAET: Not Installed
            ZMQ: 3.2.2
           Mako: Not Installed
```

PS: also hitting #16409 on the same setup. Otherwise, unrelated as it's a SNI issue, but I don't know if HTTPS URLs are handled differently by SaltStack.

**Comment by pitatus** [comment]
has this worked in the past (what version), were you doing anything different (different state file contents)?  Can you provide an example of what was working, and the version used at the time?

**Comment by SaltwaterC** [comment]
It used to work flawlessly until 2014.7.0 (I think). Also, it's a state that's used by virtually every machine as that repo is an internal repo used for our own builds. The state wasn't modified since `Thu Feb 20 16:01:07 2014 +0200` per git log and I'm fairly sure that I was using the latest SaltStack at that moment. Besides, I am upgrading fairly often to the latest version by using the SaltStack PPA.

I don't know if a previous version broke things since I was mostly out of the office for the past couple of months, but this week when I started to add machines, I saw this issue. Last time when it successfully worked was on 8th of September, 2014, but I haven't added new machines until this week.
